hub_app/test/create_post/view/create_post_view_test.dart

```diff
@@ -0,0 +1,152 @@
+    implements CreatePostBloc {}
+
+void main() {
+  group('CreatePostView', () {
+    late CreatePostBloc bloc;
+
+    void mockState(CreatePostState state) {
+      whenListen(
+        bloc,
+        Stream.fromIterable([state]),
+        initialState: state,
+      );
+    }
+
+    setUp(() {
+      bloc = _MockCreatePostBloc();
+      mockState(const CreatePostState());
+    });
+
+    testWidgets('renders correctly', (tester) async {
+      await tester.pumpsubject(createPostBloc: bloc);
+
+      expect(find.byType(CreatePostView), findsOneWidget);
+    });
+
+    testWidgets('adds CreatePostSubmitted on confirming', (tester) async {
+      await tester.pumpsubject(createPostBloc: bloc);
+
+      await tester.enterText(find.byType(TextFormField), 'TEST');
+      await tester.tap(find.text('Confirm'));
+
+      verify(() => bloc.add(CreatePostSubmitted('TEST'))).called(1);
+    });
+
+    testWidgets('navigates back when cancelling', (tester) async {
+      final navigator = MockNavigator();
+      await tester.pumpsubject(
+        createPostBloc: bloc,
+        mockNavigator: navigator,
+      );
+
+      await tester.tap(find.text('Cancel'));
+
+      verify(navigator.pop).called(1);
+    });
+
+    testWidgets('shows snackbar when the creation works', (tester) async {
+      whenListen(
+        bloc,
+        Stream.fromIterable(
+          [
+            const CreatePostState(),
+            const CreatePostState(
+              status: CreatePostStatus.success,
+              post: Post(
+                id: '1',
+                message: '',
+                userId: '',
+              ),
+            ),
+          ],
+        ),
+        initialState: const CreatePostState(),
+      );
+      await tester.pumpsubject(createPostBloc: bloc);
+      await tester.pump();
+
+      expect(find.text('Post created'), findsOneWidget);
+    });
+
+    testWidgets('shows is empty message', (tester) async {
+      mockState(
+        const CreatePostState(
+          status: CreatePostStatus.failure,
+          failure: CreatePostFailure.empty,
+        ),
+      );
+      await tester.pumpsubject(createPostBloc: bloc);
+
+      expect(find.text('Message is empty'), findsOneWidget);
+    });
+
+    testWidgets('shows too long message', (tester) async {
+      mockState(
+        const CreatePostState(
+          status: CreatePostStatus.failure,
+          failure: CreatePostFailure.tooLong,
+        ),
+      );
+      await tester.pumpsubject(createPostBloc: bloc);
+
+      expect(find.text('Message is too long'), findsOneWidget);
+    });
+
+    testWidgets('shows no profane words message', (tester) async {
+      mockState(
+        const CreatePostState(
+          status: CreatePostStatus.failure,
+          failure: CreatePostFailure.isProfane,
+        ),
+      );
+      await tester.pumpsubject(createPostBloc: bloc);
+
+      expect(find.text("Please, don't include profane words"), findsOneWidget);
```

Comment:
Wouldn't you use the l10n values for these strings in your tests since you are using those values in your code?

Reply:
I usually prefer to use the plain string instead of the value in the `l10n`. Yeah, that will make the test more fragile, but I feel it makes the test safer.

I don't have a strong opinion on this though, and I don't think we have a standard settle for this, so happy to change if you would prefer me to.

Reply:
I usually prefer to use the l10n strings in tests to avoid duplicating strings and making the tests less fragile, and to also avoid localization issues. I'm not as worried about this project since it will most likely be english only.

Reply:
Alright, as discussed in slack, I am going to keep as is since there are more instances like this in the project, then, I will update all of them in a single PR